rpc/query_server.go

```diff
@@ -764,13 +764,8 @@ func (s *QueryServer) EthGetTransactionReceipt(hash eth.Data) (resp eth.JsonTxRe
 		return resp, nil
 	}
 	if err != nil {
-		resp, err = getReceiptByTendermintHash(snapshot, s.BlockStore, r, txHash)
-		if err != nil {
-			// return empty response if cannot find hash
-			resp.Status = eth.EncInt(int64(StatusTxFail))
-			return resp, nil
-		}
-		return resp, err
+		resp, _ = getReceiptByTendermintHash(snapshot, s.BlockStore, r, txHash)
```

Comment:
If you're just going to ignore the error from `getReceiptByTendermintHash` then there's no point in `getReceiptByTendermintHash` returning an error at all. Furthermore, in `getReceiptByTendermintHash` if `blockStore.GetTxResult(hash)` doesn't error out then you can look at `txResults.Result.Code` to see if the tx succeeded or not and set the tx receipt status to match.

Reply:
Moved the error ignoring into `getReceiptByTendermintHash` as there are some errors that should probably not be ignored.